### PR TITLE
chore(ui-react): lint primitives (P-S)

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Lint all React UI primitives and related files between `P` and `S`

#### Issue #, if available
NA

#### Description of how you validated changes
`yarn react lint && yarn react test`
#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by wlee221 — DISMISSED** [review]

File: packages/react/.eslintrc.js [hunk 1]
```diff
@@ -7,6 +9,38 @@ const sharedExtensions = [
   'prettier',
 ];
 
+const primtivePatterns = templateJoin(
```
Comment (reesscot) [inline packages/react/.eslintrc.js:12]:
```suggestion
const primitivePatterns = templateJoin(
```

File: packages/react/src/primitives/StepperField/StepperField.tsx [hunk 2]
```diff
@@ -26,8 +26,6 @@ const StepperFieldPrimitive: Primitive<StepperFieldProps, 'input'> = (
   const {
     className,
     descriptiveText,
-    // this is only required in useStepper hook but deconstruct here to remove its existence in rest
-    defaultValue,
```
Comment (reesscot) [inline packages/react/src/primitives/StepperField/StepperField.tsx:30]: Same question here, I think this probably causes an error if passed through.

File: packages/react/src/primitives/SliderField/SliderField.tsx [hunk 3]
```diff
@@ -43,7 +43,6 @@ const SliderFieldPrimitive: Primitive<SliderFieldProps, typeof Root> = (
     testId,
     thumbColor,
     trackSize,
-    dir,
```
[the remark this answers is not in the document]
Comment (calebpollman) [reply packages/react/src/primitives/SliderField/SliderField.tsx:46]: It was unused, so technically it's no longer being thrown away 😄

File: packages/react/src/primitives/SwitchField/useSwitch.ts [hunk 4]
```diff
@@ -20,7 +28,10 @@ export const useSwitch = (props: UseSwitchProps) => {
         return;
       }
 
-      typeof onChange === 'function' && onChange(event);
+      if (typeof onChange === 'function') {
```
Comment (zchenwei) [inline packages/react/src/primitives/SwitchField/useSwitch.ts:31]: We have a `isFunction` util under `shared` folder

File: packages/react/src/primitives/SearchField/__tests__/SearchField.test.tsx [hunk 5]
```diff
@@ -70,23 +70,21 @@ describe('SearchField component', () => {
   });
 
   it('should forward callback ref to DOM element', async () => {
-    let ref: HTMLInputElement;
-
-    const setRef = (node) => (ref = node);
+    const ref = React.createRef<HTMLInputElement>();
```
Comment (zchenwei) [inline packages/react/src/primitives/SearchField/__tests__/SearchField.test.tsx:73]: I think using the callback ref is intentional here if you look at the test name

File: packages/react/src/primitives/StepperField/useStepper.ts [hunk 6]
```diff
@@ -62,25 +88,37 @@ export const useStepper = ({
...
 
   const handleOnChange: React.ChangeEventHandler<HTMLInputElement> =
-    React.useCallback((event) => {
-      setInputValue(event.target.value);
-    }, []);
+    React.useCallback(
+      (event) => {
+        setInputValue(event.target.value);
+
+        if (typeof onChange === 'function') {
```
Comment (zchenwei) [inline packages/react/src/primitives/StepperField/useStepper.ts:98]: We have a `isFunction` util under `shared` folder

File: packages/react/src/primitives/SearchField/__tests__/SearchField.test.tsx (hunk 5 above)
Comment (calebpollman) [reply packages/react/src/primitives/SearchField/__tests__/SearchField.test.tsx:73]: The `createRef` API was added in v16.3. Callback refs are no longer recommended in React, and while I am all for testing for edge cases, testing for functionality that has been considered legacy since versions of React below our supported version feels extraneous, not to mention potentially problematic if React was to fully drop support for callback refs

File: packages/react/src/primitives/SwitchField/useSwitch.ts (hunk 4 above)
Comment (calebpollman) [reply packages/react/src/primitives/SwitchField/useSwitch.ts:31]: Really trying not to get in to refactoring in these PRs to prevent scope creep from kicking in

**Review by zchenwei — APPROVED** [review]

File: packages/react/src/primitives/StepperField/useStepper.ts [hunk 7]
```diff
@@ -62,25 +88,37 @@ export const useStepper = ({
...
+          onChange(event);
```
Comment (calebpollman) [inline packages/react/src/primitives/StepperField/useStepper.ts:99]: Just a note that this is a bug fix, previous to this PR `onChange` props passed to `StepperField` was never called

File: packages/react/src/primitives/types/stepperField.ts [hunk 8]
```diff
@@ -57,9 +57,7 @@ export interface StepperFieldProps extends TextFieldProps {
 
   /**
    * @description
-   * TODO:
-   * Extends StepperField props from Omit<TextFieldProps, 'onChange'>, after removing [key: string]: any from the base type
-   * and rename onStepChange to onChange
```
Comment (calebpollman) [inline packages/react/src/primitives/types/stepperField.ts:62]: This was being rendered in the docs props tables

File: packages/react/src/primitives/StepperField/useStepper.ts (hunk 7 above)
Comment (reesscot) [reply packages/react/src/primitives/StepperField/useStepper.ts:99]: Good catch

File: packages/react/src/primitives/types/stepperField.ts (hunk 8 above)
Comment (reesscot) [reply packages/react/src/primitives/types/stepperField.ts:62]: 🤦‍♂️

File: packages/react/src/primitives/SliderField/SliderField.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (reesscot) [reply packages/react/src/primitives/SliderField/SliderField.tsx:46]: Yeah, that should have been getting passed through, so your change looks correct:
https://www.radix-ui.com/docs/primitives/components/slider#root

**Review by reesscot — APPROVED** [review]
LGTM, thanks for catching these issues
